lua/lspconfig/server_configurations/sumneko_lua.lua

```diff
@@ -33,6 +33,7 @@ return {
       return util.find_git_ancestor(fname)
     end,
     single_file_support = true,
+    new_folder_restart = true,
```

Comment:
why is it optional? also need docs update

Reply:
because if server support dynamically add or remove workspace Folders . no need restart server. just send notify `didChangeWorkspaceFolders`.

Reply:
if only sumneko needs this then it's a sumneko bug, not something we should add code to workaround.

where is this `return` structure documented? adding random fields to it in random lang servers, will lead to a fragile codebase.

Reply:
I agree with you. But it's hard to pin the problem down to a bug in sumnkeolua. The lsp document just stipulates the rules of the communication protocol. Does not specify any details of the implementation. So you can see that each server works differently in some regard. And the author has indicated in the wiki that dynamic addition or deletion of working folders is not supported. The pr just does a behavior similar to vscode.

Reply:
I've opened an issue at the repository, but there's no response yet.
What about hardcoding the restart behavior for sumneko in a way that is easy to remove later on, in case there's a fix?
So far only sumneko behaves like this, so it would be weird to expose it as a config. I also can't imagine a user who would prefer the alternative option, since it behaves incorrectly.

Reply:
I don't recommend classifying this as a language server bug. This is somewhat rude to the language server authors. The lsp does not specify how language server authors are to implement it. Depending on the author's code level, the language server behaves differently. Sumneko has described the current workspacefolder behavior in the wiki. This is a must if we want to support this server at the moment. It would be great if sumneko could be dynamically added or removed in the future.

about  this option there are many language servers. I didn't check their docs fully and maybe there has other servers that needs to be restarted. currently as far as I know is sumneko , but be careful. not every language server pays attention to `workspacefolders`, for example clang doesn't actually depend on `workspacefolders`.